### PR TITLE
except +

### PR DESCRIPTION
**Pull request by scopatz**
Python interface will now raise exceptions when it encounters an error in C++ or Cython. 

Should address #1463, and if not it at least addresses some other, similar cases.

CC @bam241 @FlanFlanagan

### CODE REVIEW

File: cyclus/agents.pxd [hunk 1]
```diff
@@ -17,22 +17,22 @@ from cyclus.cpp_cyclus cimport shared_ptr
...
-    cpp_cyclus.Inventories SnapshotInv()
-    std_string schema()
-    cpp_jsoncpp.Value annotations()
-    void Build(cpp_cyclus.Agent*)
-    void EnterNotify()
-    void BuildNotify()
-    void DecomNotify()
-    void AdjustMatlPrefs(cpp_cyclus.PrefMap[cpp_cyclus.Material].type&)
-    void AdjustProductPrefs(cpp_cyclus.PrefMap[cpp_cyclus.Product].type&)
+    std_string version()  except +
```
Comment (bam241) [inline cyclus/agents.pxd]: there is 2 spaces, there

File: src/eventhooks.pxd [hunk 2]
```diff
@@ -1,3 +1,3 @@
 """Header for cyclus events."""
 
-cdef public void eventloophook "CyclusEventLoopHook" ()
+cdef public void eventloophook "CyclusEventLoopHook" ()  except +
```
Comment (bam241) [inline src/eventhooks.pxd]: 2 spaces there as well

File: src/eventhooks.pyx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 """The event handling module in cyclus."""
 from __future__ import print_function, unicode_literals
 
-cdef public void eventloophook "CyclusEventLoopHook" ():
+cdef public void eventloophook "CyclusEventLoopHook" ()  except +:
```
Comment (bam241) [inline src/eventhooks.pyx]: 2 spaces there too

**Comment by scopatz** [comment]
Thanks @bam241, I have fixed the spacing issues.

**Comment by bam241** [comment]
works but don't fix the #1463 issue

**Comment by bam241** [comment]
thx @scopatz !

**Pull request merged**